app/controllers/materials_controller.rb

```diff
@@ -37,4 +53,12 @@ def send_sheet(sheet)
   def search_params
     params.permit(:material_name, :material_function, :manufacturer_id, :numerical_filter, :min, :max)
   end
+
+  def add_remove_compare
+    if session[:compare].include?(params[:id])
+      session[:compare].delete(params[:id])
+    else
+      session[:compare].append(params[:id])
```

Comment:
Might be nice to extract params[:id] into a variable or a method. Same for session[:compare].